ingester/plugins/iot_plugin.go

```diff
@@ -43,6 +43,60 @@ var lookup = map[string]func(msg xbospb.XBOS) (float64, bool){
 		}
 		return 0, false
 	},
+	"demand": func(msg xbospb.XBOS) (float64, bool) {
+		if has_meter(msg) && msg.XBOSIoTDeviceState.Meter.Demand != nil {
+			return float64(msg.XBOSIoTDeviceState.Meter.Demand.Value), true
+		}
+		return 0, false
+	},
+	"reactive_power": func(msg xbospb.XBOS) (float64, bool) {
+		if has_meter(msg) && msg.XBOSIoTDeviceState.Meter.ReactivePower != nil {
+			return float64(msg.XBOSIoTDeviceState.Meter.ReactivePower.Value), true
+		}
+		return 0, false
+	},
+	"current": func(msg xbospb.XBOS) (float64, bool) {
+		if has_meter(msg) && msg.XBOSIoTDeviceState.Meter.Current != nil {
+			return float64(msg.XBOSIoTDeviceState.Meter.Current.Value), true
+		}
+		return 0, false
+	},
+	"water_total": func(msg xbospb.XBOS) (float64, bool) {
+		if has_meter(msg) && msg.XBOSIoTDeviceState.Meter.WaterTotal != nil {
+			return float64(msg.XBOSIoTDeviceState.Meter.WaterTotal.Value), true
+		}
+		return 0, false
+	},
+	"water_rate": func(msg xbospb.XBOS) (float64, bool) {
+		if has_meter(msg) && msg.XBOSIoTDeviceState.Meter.WaterRate != nil {
+			return float64(msg.XBOSIoTDeviceState.Meter.WaterRate.Value), true
+		}
+		return 0, false
+	},
+	"water_instantaneous": func(msg xbospb.XBOS) (float64, bool) {
+		if has_meter(msg) && msg.XBOSIoTDeviceState.Meter.WaterInstantaneous != nil {
+			return float64(msg.XBOSIoTDeviceState.Meter.WaterInstantaneous.Value), true
+		}
+		return 0, false
+	},
+	"condense_total": func(msg xbospb.XBOS) (float64, bool) {
+		if has_meter(msg) && msg.XBOSIoTDeviceState.Meter.CondenseTotal != nil {
+			return float64(msg.XBOSIoTDeviceState.Meter.CondenseTotal.Value), true
+		}
+		return 0, false
+	},
+	"condense_rate": func(msg xbospb.XBOS) (float64, bool) {
+		if has_meter(msg) && msg.XBOSIoTDeviceState.Meter.CondenseRate != nil {
+			return float64(msg.XBOSIoTDeviceState.Meter.CondenseRate.Value), true
+		}
+		return 0, false
+	},
+	"condense_instantaneous": func(msg xbospb.XBOS) (float64, bool) {
+		if has_meter(msg) && msg.XBOSIoTDeviceState.Meter.CondenseInstantaneous != nil {
+			return float64(msg.XBOSIoTDeviceState.Meter.CondenseInstantaneous.Value), true
+		}
+		return 0, false
+	},
```

Comment:
This is a great start! Can you add some default units to the plugin as well? https://github.com/gtfierro/xboswave/blob/71e4f807b87e2350b275b3c882aeec0fae18f7f7/ingester/plugins/iot_plugin.go#L188

Reply:
Oh yes, makes sense, I will create new key-value pairs mapping the new meter fields to default units :)

Reply:
Resolved. Two follow ups:
1. I added units for 8 of the new fields (water/condense instantaneous/rate/total, demand, reactive power). The `current` field already had an entry in the units map.
2. The units for the water/condense instantaneous/rate is "gallons per minute" which I abbreviated as gpm. I'm not sure whether to write out the unit explicitly, or keep the abbreviation instead.

Reply:
`gpm` is fine for now